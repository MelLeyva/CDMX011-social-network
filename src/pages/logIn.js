export const logIn = () => {
  const view = `
    <div>
        <h1>soy el log In </h1>
    </div>
    `;
  return view;
};
