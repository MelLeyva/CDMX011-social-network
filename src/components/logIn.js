/* eslint-disable import/no-cycle */
import { logInWhitGoogle, githubSignin } from '../lib/logInFb.js';

// eslint-disable-next-line import/no-cycle
import { onNavigate } from '../main.js';

export const logIn = () => {
  document.body.style.backgroundImage = 'url(../images/signUp.png)';
  const theBody = document.querySelector('body');
  const divLogIn = document.createElement('section');
  const tagNav = document.createElement('nav');
  const tagUl = document.createElement('ul');
  const tagLi = document.createElement('li');
  const tagHrefReturnHome = document.createElement('a');
  const tagI = document.createElement('i');
  const tagTitle = document.createElement('h1');
  const formulario = document.createElement('form');
  const labelEmail = document.createElement('label');
  const inputEmail = document.createElement('input');
  const labelPassword = document.createElement('label');
  const inputPassword = document.createElement('input');
  const buttonLogIn = document.createElement('button');
  // const buttonReturn = document.createElement('button');
  const tagOr = document.createElement('h4');
  const buttonGoogle = document.createElement('button');
  const divLoginGoogle = document.createElement('div');
  const spanDivLoginGoogle = document.createElement('span');
  const imagenButton = document.createElement('img');
  const btnGitLogIn = document.createElement('button');
  const divLogInGit = document.createElement('div');
  const spanBtnLogInGit = document.createElement('span');
  const imgGitLogIn = document.createElement('img');

  theBody.classList.add('login-body');
  divLogIn.classList.add('div_login');
  tagI.classList.add('fas', 'fa-home');
  tagTitle.classList.add('title_login');
  formulario.classList.add('sub_contenedor');
  buttonLogIn.classList.add('btn_login');
  tagOr.classList.add('tag_or');
  buttonGoogle.classList.add('btn_google');
  btnGitLogIn.classList.add('btn_git');

  tagHrefReturnHome.setAttribute('href', '#');
  inputEmail.setAttribute('type', 'text');
  inputPassword.setAttribute('type', 'password');
  imagenButton.setAttribute('src', '../images/logo_google.png');
  imgGitLogIn.setAttribute('src', '../images/github.png');

  tagTitle.textContent = 'Iniciar sesion';
  tagOr.textContent = '-o-';
  labelEmail.textContent = 'Correo electronico';
  labelPassword.textContent = 'ContraseÃ±a';
  buttonLogIn.textContent = 'Iniciar sesion';
  spanDivLoginGoogle.textContent = 'Ingresa con ';
  spanBtnLogInGit.textContent = 'Ingresa con ';
  // icono home, funcion retorno a home
  tagHrefReturnHome.addEventListener('click', (e) => {
    e.preventDefault();
    onNavigate('/');
  });

  buttonGoogle.addEventListener('click', (e) => {
    e.preventDefault();
    logInWhitGoogle();
  });

  btnGitLogIn.addEventListener('click', (e) => {
    e.preventDefault();
    githubSignin();
  });

  tagNav.appendChild(tagUl);
  tagUl.appendChild(tagLi);
  tagLi.appendChild(tagHrefReturnHome);
  tagHrefReturnHome.appendChild(tagI);

  divLogIn.appendChild(tagNav);
  divLogIn.appendChild(tagTitle);
  divLogIn.appendChild(formulario);

  formulario.appendChild(labelEmail);
  formulario.appendChild(inputEmail);
  formulario.appendChild(labelPassword);
  formulario.appendChild(inputPassword);
  formulario.appendChild(buttonLogIn);
  formulario.appendChild(tagOr);
  formulario.appendChild(buttonGoogle);
  formulario.appendChild(btnGitLogIn);

  buttonGoogle.appendChild(divLoginGoogle);
  divLoginGoogle.appendChild(spanDivLoginGoogle);
  divLoginGoogle.appendChild(imagenButton);
  btnGitLogIn.appendChild(divLogInGit);
  divLogInGit.appendChild(spanBtnLogInGit);
  divLogInGit.appendChild(imgGitLogIn);

  return divLogIn;
};
