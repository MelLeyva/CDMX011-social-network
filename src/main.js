/* eslint-disable import/no-cycle */
import { home } from './components/home.js';
// eslint-disable-next-line import/no-cycle
import { logIn } from './components/logIn.js';
import { signUp } from './components/signUp.js';

export const routes = {
  '/': home,
  '/logIn': logIn,
  '/signUp': signUp,
};

export const onNavigate = (pathname) => {
  const rootDiv = document.getElementById('root');
  window.history.pushState(
    {},
    pathname,
    window.location.origin + pathname,
  );
  while (rootDiv.firstChild) {
    rootDiv.removeChild(rootDiv.firstChild);
  }
  rootDiv.appendChild(routes[window.location.pathname]());
};

window.onpopstate = () => {
  const rootDiv = document.getElementById('root');
  while (rootDiv.firstChild) {
    rootDiv.removeChild(rootDiv.firstChild);
  }
  rootDiv.appendChild(routes[window.location.pathname]());
};

export const render = () => {
  const rootDiv = document.createElement('div');
  window.onpopstate();
  // console.log(window.onpopstate());
  return rootDiv;
};
