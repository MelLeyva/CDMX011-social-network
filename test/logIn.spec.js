/**
 * @jest-environment jsdom
 */

import './globals/fibase.js';
import { render } from '../src/main.js';
import { logIn } from '../src/components/logIn.js';
// import { loginWithEmail } from '../src/lib/logInFb.js';

describe('Clic en el boton logIn', () => {
  let rootdiv = document.createElement('div');
  rootdiv.setAttribute('id', 'root');
  document.body.appendChild(rootdiv);
  render();
  document.querySelector('.btn_log_in').click();
  rootdiv = document.getElementById('root');
  test('Test sobre log In', () => {
    const form = document.querySelector('.sub_contenedor');
    expect(form.outerHTML).toMatchSnapshot();
  });

  const fillOutAndSubmitForm = (email, password) => {
    document.getElementById('user').value = email;
    document.getElementById('pass').value = password;
    document.querySelector('.btn_log_in').click();
  };

  it('login al usuario', async () => {
    const email = 'test@prueba.com';
    const password = '12345678';
    const user = { email, uid: 'xxxxxxxx' }; // creamos un objeto ficticio del usuario
    // Haremos uso de mock creando una promesa (exitosa) y retona el usuario ficticio
    const mockSignInWithEmailAndPassword = jest.fn();
    mockSignInWithEmailAndPassword.mockResolvedValue(user);

    // mokeamos los metodos de firebase.auth
    const mockFirebaseAuth = {
      signInWithEmailAndPassword: mockSignInWithEmailAndPassword,
      currentUser: user,
      onAuthStateChanged: () => {},
    };
    console.log(mockSignInWithEmailAndPassword.mock);
    // cuando firebese.auth sea invocada retornara el mock creado mockFirebaseAuth
    firebase.auth = () => mockFirebaseAuth;

    render(rootdiv, logIn());
    fillOutAndSubmitForm(email, password);

    expect(mockSignInWithEmailAndPassword).toHaveBeenCalledWith(email, password);
    expect(rootdiv.outerHTML).toContain(email);
  });
});
