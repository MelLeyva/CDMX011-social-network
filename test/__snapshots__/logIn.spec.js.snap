// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clic en el boton logIn Test sobre log In 1`] = `"<form class=\\"sub_contenedor\\"><label>Correo electronico</label><input id=\\"user\\" type=\\"email\\"><label>Contrase√±a</label><input id=\\"pass\\" type=\\"password\\"><button class=\\"btn_login\\">Iniciar sesion</button><h4 class=\\"tag_or\\">-o-</h4><button class=\\"btn_google\\"><div><span>Ingresa con </span><img src=\\"../images/logo_google.png\\"></div></button><button class=\\"btn_git\\"><div><span>Ingresa con </span><img src=\\"../images/github.png\\"></div></button></form>"`;
